bob/src/cluster/quorum.rs

```diff
@@ -351,22 +439,74 @@ impl Cluster for Quorum {
-                    exist[ind] |= r;
+        debug!("EXIST {} keys", len);
+
+        let mut result = vec![false; len];
+
+        let (local, mut primary, mut secondary) = Self::group_by_nodes(keys, &self.mapper);
+
+        if let Some(local) = local {
+            if !local.is_empty() {
+                trace!("EXIST {} keys check local node", len);
+                let local_keys = local.collect(keys);
+                match exist_on_local_node(&self.backend, &local_keys).await {
+                    Ok(local_exist) => {
+                        trace!("EXIST {} keys check local node: found {}/{} keys",
+                               len, local_exist.iter().filter(|v| **v).count(), local.len());
+                        local.update_existence(&mut result, &local_exist);
+                    },
+                    Err(e) => warn!("EXIST {} check local node failed: {:?}", len, e)
+                };
+            }
+        }
+
+        trace!("EXIST {} keys check primary nodes", len);
+        Self::collect_remote_exists(&mut result, keys, &mut primary).await;
+        trace!("EXIST {} keys check primary nodes finished", len);
+
+        trace!("EXIST {} keys check secondary nodes", len);
+        Self::collect_remote_exists(&mut result, keys, &mut secondary).await;
+        trace!("EXIST {} keys check secondary nodes finished", len);
+
+        let mut alien_index_map = IndexMap::where_not_exists(&result);
+
+        if !alien_index_map.is_empty() {
+            trace!("EXIST {} keys check local alien", len);
+            match exist_on_local_alien(&self.backend, &alien_index_map.collect(keys)).await {
+                Ok(local_alien_result) => {
+                    trace!("EXIST {} keys check local alien finished: found {}/{} keys",
+                           len, local_alien_result.iter().filter(|v| **v).count(), alien_index_map.len()); 
+                    alien_index_map.update_existence(&mut result, &local_alien_result);
+                    alien_index_map.retain_not_existed(&result)
+                },
+                Err(e) => warn!("EXIST {} keys check local alien failed: {:?}", len, e)
+            }
+        }
+
+        if !alien_index_map.is_empty() {
+            trace!("EXIST {} keys check remote alien", len);
+            let all_remote_nodes: Vec<_> = self
+                .mapper
+                .nodes()
+                .iter()
+                .filter(|n| n.name() != self.mapper.local_node_name())
+                .collect();
+            let remote_nodes_aliens_exist =
+                exist_on_remote_aliens(&all_remote_nodes, &alien_index_map.collect(keys)).await;
+            trace!("EXIST {} keys check remote alien finished", len);
+            for remote_alien_result in remote_nodes_aliens_exist {
+                match remote_alien_result {
+                    Ok(remote_alien_result) => {
+                        alien_index_map.update_existence(&mut result, remote_alien_result.inner());
```

Comment:
Please, add `trace` log here, printing the number of keys found on node (`EXIST in aliens on node {}: found {}/{} keys`)